### ISSUE TITLE
pypandoc fails with: ValueError: stat: path too long for Windows

### ISSUE THREAD

**Issue by MissingNoIOI** [bebraw/pypandoc#54]
Hi, 
I wrote a small script using pypandoc, by passing it a preformatted string, and it works like a charm on Linux, but when I tried it on Windows it fails with the following Traceback:

```
Traceback (most recent call last):
  File ".\PycharmProjects\reddit2ebook\reddit2ebook\main.py", line 101, in <module>
    main()
  File ".\PycharmProjects\reddit2ebook\reddit2ebook\main.py", line 45, in main
    create_ebook(text, directory, bookname)
  File ".\PycharmProjects\reddit2
ebook\reddit2ebook\main.py", line 96, in create_ebook
    outputfile=directory + ebook_name + '.epub')
  File "C:\Python34\lib\site-packages\pypandoc.py", line 94, in convert
    outputfile=outputfile, filters=filters)
  File "C:\Python34\lib\site-packages\pypandoc.py", line 99, in _convert
    source, format = reader(source, format, encoding=encoding)
  File "C:\Python34\lib\site-packages\pypandoc.py", line 140, in _read_file
    path = os.path.exists(source)
  File "C:\Python34\lib\genericpath.py", line 19, in exists
    os.stat(path)
ValueError: stat: path too long for Windows
```

It seems that _read_file  tries to call os.path.exists() with the source to check if its a the path to a valid file or a string containing formatted text. While this works on *nix Systems, it fails on Windows due to the limit of 255 characters per path, which most texts easily surpass.

For now I could fix it by including:

```
    except ValueError:
        path = ''
```
in line 142 after the try catch block

**Comment by bebraw** [comment]
Yeah, I can see that fix would work here. Before doing that let's wait for a little more feedback. There are some guys in the know following. :+1:

**Comment by xysmas** [comment]
There might be a more elegant workaround using something like the 

    \\?\

[found here](https://msdn.microsoft.com/en-us/library/aa365247(v=vs.85).aspx#maxpath), but I admit, this is far out of my normal domain. 

The easiest solution is to stop using windows.

**Comment by MissingNoIOI** [comment]
I personally don't use it, but I uploaded the script to PyPi and thought it might be useful for it to work on all operating systems. And it was somwhow foreseeable that Windows is the only one making problems :/

**Comment by bebraw** [comment]
I applied the fix you proposed. It makes perfect sense on Windows.

Looks like our Travis build has gone broken, though. I think their infrastructure changed. Maybe that has something to do with it? See #55. Once that has been resolved I'll publish a new version.

In the meantime you can consider using standalone version of pypandoc as I have no idea how long it will take to resolve that.

**Comment by bebraw** [comment]
@MissingNoIOI FYI, I published `0.9.8` with this fix. Travis is broken but we can live with that.

**Comment by gamicklisafisher** [comment]
You should check out ‘Long Path Tool’ it will help you fix the issues.
I had faced similar issue and this tool helped me to resolve this issue.
